File: frontend/src/types/actionLog.ts
export interface ActionLog {
  id: string

  event_type: string
  entity_type: string
  entity_id: string | null

  message: string

  details: Record<string, unknown>

  created_at: string
}

export interface ActionLogFilters {
  event_type?: string
  entity_type?: string
  limit?: number
  offset?: number
}

File: backend/app/schemas/action_log.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import (
    BaseModel,
    ConfigDict,
)


class ActionLogRead(BaseModel):
    model_config = ConfigDict(
        from_attributes=True,
    )

    id: UUID

    event_type: str
    entity_type: str
    entity_id: UUID | None

    message: str

    details: dict[str, Any]

    created_at: datetime